transactor/src/main/scala/io/mediachain/transactor/CborSerialization.scala

```diff
@@ -0,0 +1,479 @@
+    *        `typeNames` set
+    */
+  def assertOneOfRequiredTypeNames(cMap: CMap, typeNames: Set[CborTypeName])
+  : Xor[DeserializationError, Unit] = {
+    getTypeName(cMap)
+      .flatMap { name =>
+        if (typeNames.contains(name)) {
+          Xor.right({})
+        } else {
+          Xor.left(UnknownObjectType(name.toString))
+        }
+      }
+  }
+
+  /**
+    * Get the value of the `type` field from the given cbor map
+ *
+    * @param cMap a cbor `CMap` to get the type name from
+    * @return the value of the `type` field, or a `DeserializationError` error if
+    *         no `type` field exists, or its value is not a valid type name
+    */
+  def getTypeName(cMap: CMap): Xor[DeserializationError, CborTypeName] =
+    getRequired[CString](cMap, "type")
+      .map(_.string)
+      .leftMap(_ => TypeNameNotFound())
+      .flatMap(CborTypeNames.fromString)
+
+
+  /**
+    * Get the value of a required field in a cbor map.
+ *
+    * @param cMap a cbor `CMap` to pull the field from
+    * @param fieldName the name of the field
+    * @tparam T the type of `CValue` to return
+    * @return the value of the field, or a `RequiredFieldNotFound` error if
+    *         the field doesn't exist
+    */
+  def getRequired[T <: CValue](cMap: CMap, fieldName: String)
+  : Xor[RequiredFieldNotFound, T] = Xor.fromOption(
+    cMap.getAs[T](fieldName),
+    RequiredFieldNotFound(fieldName)
+  )
+
+
+  /**
+    * Get a required field whose value is an encoded `Reference` type.
+ *
+    * @param cMap a cbor `CMap` to pull the reference field from
+    * @param fieldName the name of the field
+    * @return the deserialized `Reference`, or a `DeserializationError` if
+    *         the field doesn't exist or can't be decoded as a reference
+    */
+  def getRequiredReference(cMap: CMap, fieldName: String)
+  : Xor[DeserializationError, Reference] =
+    getRequired[CMap](cMap, fieldName)
+    .flatMap(referenceFromCMap)
+
+  /**
+    * Get an optional `Reference` value from the cbor map.
+ *
+    * @param cMap a cbor `CMap` to pull the reference field from.
+    * @param fieldName the name of the field
+    * @return Some[Reference] if the field exists and can be decoded,
+    *         None otherwise
+    */
+  def getOptionalReference(cMap: CMap, fieldName: String): Option[Reference] =
+    cMap.getAs[CMap](fieldName)
+      .flatMap(referenceFromCMap(_).toOption)
+
+
+  /**
+    * Try to decode a `Reference` from a cbor map
+ *
+    * @param cMap the cbor `CMap` to decode as a `Reference`
+    * @return the decoded `Reference`, or a `DeserializationError` if decoding
+    *         fails
+    */
+  def referenceFromCMap(cMap: CMap): Xor[DeserializationError, Reference] =
+    getRequired[CMap](cMap, "@link")
+      .flatMap { linkVal: CValue =>
+        linkVal match {
+          case CString(s) if s.startsWith("dummy@") => {
+            DummyReferenceDeserializer.fromCMap(cMap)
```

Comment:
Well.... we could still use an integer sequence number by just stuffing it into the `meta` map and ignore the actual value of the `@link` field... but that would be hidden in the `DummyStore` implementation.

Or we could just use multihashes for the dummy store also, which I kind of prefer, since it's closer to the real implementation.